doc/api/figure_api.rst

```diff
@@ -5,5 +5,311 @@
 .. currentmodule:: matplotlib.figure
 
 .. automodule:: matplotlib.figure
-   :members:
-   :inherited-members:
+   :no-members:
+   :no-undoc-members:
+
+The Figure class
+================
+.. autosummary::
+   :toctree: _as_gen
+   :template: autosummary_class_only.rst
+   :nosignatures:
+
+   Figure
```

Comment:
I see what you are saying, but I'm not convinced it is a huge problem:

`` `matplotlib.figure` `` is the right way to get the overall index if that is what we want in the docs.

`` `.Figure` `` takes us to the instantiation docs, which I think we do _not_ want in the overview.

At least on non-mobile, the left-hand navigation _and_ the breadcrumb is pretty clear where the module level description is:

<img width="1274" alt="fig" src="https://github.com/matplotlib/matplotlib/assets/1562854/6f188b05-d34a-4dc2-9c69-8ae910474f24">

An alternative may be that we make entries for matplotlib.figure.Figure and matplotlib.figure.SubFigure.  We don't do this with any other APIs currently, but I don't think there is any reason they API reference has to slavishly correspond to files in the library directory.  I think this would mean `figure_Figure_api.rst` and `figure_SubFigure_api.rst`, but that doesn't seem the end of the world.  

Finally, an even more radical suggestions would be to rearrange the source code itself; eg have a subfigure module, and split `FigureBase` into its own private module.  That doesn't seem the end of the world - I doubt many people are doing `import matplotlib.figure.SubFigure` yet.  

So, I think the proposed change here is better than what we currently have, but I'm open to relatively major surgery if we can get something even better.

Reply:
Thinking about this more though: even if we do the more radical suggestions above, I don't see how we get the init docs onto their own page and out of the intro.  

Your suggestion to sphinx seems reasonable.  Not sure how we emulate that internally.

Reply:
One could try to separate class and init docstring, put the class on the overview page and init in subpage where the class currently is. May be possible through appropriate configuration https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#configuration.

Reply:
I think a fair bit is possible with templates as well.

Reply:
See newest version...